src/appleseed.shaders/src/appleseed/as_ray_switch.osl

```diff
@@ -0,0 +1,155 @@
+        string label = "Light Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_shadow = color(0)
+    [[
+        string as_maya_attribute_name = "colorShadow",
+        string as_maya_attribute_short_name = "csh",
+        string label = "Shadow Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_transparency = color(0)
+    [[
+        string as_maya_attribute_name = "colorTransparency",
+        string as_maya_attribute_short_name = "ctr",
+        string label = "Transparency Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_probe = color(0)
+    [[
+        string as_maya_attribute_name = "colorProbe",
+        string as_maya_attribute_short_name = "cpr",
+        string label = "Probe Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_diffuse = color(0)
+    [[
+        string as_maya_attribute_name = "colorDiffuse",
+        string as_maya_attribute_short_name = "cde",
+        string label = "Diffuse Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_glossy = color(0)
+    [[
+        string as_maya_attribute_name = "colorGlossy",
+        string as_maya_attribute_short_name = "cgl",
+        string label = "Glossy Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_specular = color(0)
+    [[
+        string as_maya_attribute_name = "colorSpecular",
+        string as_maya_attribute_short_name = "csp",
+        string label = "Specular Ray Color",
+        string page = "Color"
+    ]],
+    color in_color_subsurface = color(0)
+    [[
+        string as_maya_attribute_name = "colorSubsurface",
+        string as_maya_attribute_short_name = "csu",
+        string label = "Subsurface Ray Color",
+        string page = "Color"
+    ]],
+
+    output color out_color = color(0)
+    [[
+        string as_maya_attribute_name = "outColor",
+        string as_maya_attribute_short_name = "oc",
+        string label = "Output Color"
+    ]]      
+)
+{
+    if (raytype("camera"))
+    {
+        out_color = in_color;
+    }
+    else if (raytype("light"))
+    {
+        out_color = in_color_light;
+    }
+    else if (raytype("shadow"))
+    {
+        out_color = in_color_shadow;
+    }
+    else if (raytype("transparency"))
+    {
+        out_color = in_color_transparency;
+    }
+    else if (raytype("probe"))
+    {
+        out_color = in_color_probe;
+    }
+    else if (raytype("diffuse"))
+    {
+        out_color = in_color_diffuse;
+    }
+    else if (raytype("glossy"))
+    {
+        out_color = in_color_glossy;
+    }
+    else if (raytype("specular"))
+    {
+        out_color = in_color_specular;
+    }
+    else if (raytype("subsurface"))
+    {
+        out_color = in_color_subsurface;
+    }
+    else
+    {
+#ifdef DEBUG
+        string shadername = "";
+        getattribute("shader:shadername", shadername);
+        warning("[DEBUG!]: Unknown Ray Type in %s, %s:%d\n",
```

Comment:
Is `[DEBUG!]` the "normalized" tag we use everywhere?

Reply:
@dictoon actually no, it should be ```warning("[WARNING]"...)```